clusterman/draining/kubernetes.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import colorlog
+
+logger = colorlog.getLogger(__name__)
+
+
+def kube_drain(operator_client, host):
+    operator_client.reload_state()
+    operator_client.set_node_unschedulable(host.ip)
+    operator_client.evict_pods_on_node(host.ip)
+
+
+def kube_delete_node(operator_client, host):
+    operator_client.reload_state()
```

Comment:
Do we need to reload_state() again here? In other words, is _get_agent_metadata not enough?

Reply:
`_get_agent_metadata()` reads directly from `self._nodes_by_ip` which is only updated when reload_state is called. I would say keep it there in case to avoid NotFound errors.